### ISSUE TITLE
Cyclic interaction errors sometimes show a plugin in the cycle twice

### ISSUE THREAD

**Issue by Ortham** [loot/loot#1840]
For some reason cyclic interaction errors are sometimes reported with a plugin that gets listed with a master edge going to itself. For example, `Obsidian Weathers.esp` in the screenshot below:

![image](https://user-images.githubusercontent.com/2430738/216781187-28885449-5514-4ff6-ac9c-1b0d58b07f3d.png)

I think this has only been seen with a master edge between the duplicate entries, but it's not clear why the edge is being reported: is it a bug in how the cycle is reported or recorded, or is there actually an edge there (I think the latter)? If there's actually an edge, is it because the plugin's masters are read incorrectly, or does the plugin actually list itself as one of its masters?

This may not be an actual bug, but it looks like one and it would be good to understand what's going on. It would be good to have a debug log from when this happened, and a link to where I can download the plugin in question.

**Comment by pStyl3** [comment]
`Obsidian Weathers.esp` - [Obsidian Weathers and Seasons](https://www.nexusmods.com/skyrimspecialedition/mods/12125/)
`AAE Ultimate Edition.esp` & `AAE_Fixes_Patch.esp` - [Advanced Adversary Encounters - Ultimate SSE](https://www.nexusmods.com/skyrimspecialedition/mods/6843/)
`DeadlyDragons.esp` - [Deadly Dragons](https://www.nexusmods.com/skyrimspecialedition/mods/23723/)
`Mortal Enemies.esp` - [Mortal Enemies SE](https://www.nexusmods.com/skyrimspecialedition/mods/4881/)
`Regional Sounds Expansion.esp` - [Regional Sounds Expansion](https://www.nexusmods.com/skyrimspecialedition/mods/77829/)

Using the following `userlist.yaml`:
```yaml
plugins:
  - name: 'DeadlyDragons.esp'
    after: ['AAE_Fixes_Patch.esp']
  - name: 'Mortal Enemies.esp'
    after: ['DeadlyDragons.esp']
  - name: 'Obsidian Weathers.esp'
    after: ['Regional Sounds Expansion.esp']
```
I can replicate the cyclic interaction, but I do _not_ get a duplicate `Obsidian Weathers.esp` entry in the cycle message.

**Comment by sibir-ine** [comment]
I was able to replicate the duplicate plugin issue by doing as pStyl3 did above as well as installing the True Storms patch (with or without True Storms) on the Obsidian Weathers page. Maybe it's accidentally printing the name of the patch's master when it should be printing the name of the patch? Not sure though since there shouldn't be a group edge between the patch & `AAE Ultimate Edition.esp`; they're both in default. Here's [my debug log](https://github.com/loot/loot/files/10610689/LOOTDebugLog.txt).

**Comment by pStyl3** [comment]
With `True Storms - Obsidian Weathers - Patch .esp` being present, I can replicate `Obsidian Weathers.esp` getting mentioned twice, as well.

**Comment by Ortham** [comment]
Thanks, I'm able to replicate the issue now. I can see that the log doesn't mention a master edge being added from Obsidian Weathers.esp to itself, but that the exception thrown by libloot lists the edge. Turns out it was due to the cyclic path being recorded incorrectly.

**Comment by Ortham** [comment]
@pStyl3 @sibir-ine Can you try with [this libloot build](https://loot.jfrog.io/ui/native/libloot/fix-cycle-errors/) (just drop it in a LOOT v0.19 install)? It fixes the issue for me, but it would be good to double-check.

**Comment by sibir-ine** [comment]
Yup, no duplicate entry on my end.

**Comment by pStyl3** [comment]
Can confirm, the issue is fixed for me as well.

**Comment by Ortham** [comment]
Fixed as of 80f295b922d714945f480171307c0dd7707af62a.
